### PR TITLE
EVG-12586: Create Volume Modal

### PR DESCRIPTION
**Pull request by SupaJoon**
https://jira.mongodb.org/browse/EVG-12586
The best way to test these code changes is to point your local development server to production and attempt to spawn a volume with different configurations from [this page](http://localhost:3000/spawn/volume)

### CODE REVIEW

File: src/components/Spawn/MountVolumeSelect.tsx [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+  useEffect(() => {
+    if (data?.myHosts) {
+      const opts = data.myHosts
+        // Filter hosts that do not have the same availability zone as the volume.
+        .filter(
+          ({ availabilityZone }) => availabilityZone === targetAvailabilityZone
+        )
+        // Map host to a displayName and ID for the dropdown <Option />
+        .map((host) => ({
+          key: host.id,
```
Comment (khelif96) [inline src/components/Spawn/MountVolumeSelect.tsx:59]: Why rename id to key?

File: src/components/styles/Layout.tsx [hunk 2]
```diff
@@ -63,3 +63,12 @@ export const InputLabel = styled.label`
 export const ErrorMessage = styled(Body)`
   color: ${uiColors.red.base};
 `;
+
+export const HR = styled("hr")`
+  background-color: ${uiColors.gray.light2};
```
Comment (khelif96) [inline src/components/styles/Layout.tsx:68]: nit. Can we use destructuring for this. `background-color: ${gray.light2}`

File: src/gql/queries/subnet-availability-zones.ts [hunk 3]
```diff
@@ -0,0 +1,7 @@
+import { gql } from "@apollo/client";
+
+export const GET_SUBNET_AVAILABILITY_ZONES = gql`
+  query subnetAvailabilityZones {
```
Comment (khelif96) [inline src/gql/queries/subnet-availability-zones.ts:4]: nit. capitalize subnet, The rest of our queries are capitalized

File: src/pages/spawn/spawnVolume/spawnVolumeButton/SpawnVolumeModal.tsx [hunk 4]
```diff
@@ -1,57 +1,80 @@
...
-interface Action {
-  type: ActionType;
-  data: any;
-}
+type Action =
+  | { type: "setSize"; data: number }
+  | { type: "setAvailabilityZone"; data: string }
+  | { type: "editExpiration"; expiration?: Date; noExpiration?: boolean }
+  | { type: "setHost"; hostId: string }
+  | { type: "setType"; typeId: string };
```
Comment (khelif96) [inline src/pages/spawn/spawnVolume/spawnVolumeButton/SpawnVolumeModal.tsx:43]: Lets move this out into its own hook to clean up this component a little more.

File: src/pages/spawn/spawnVolume/spawnVolumeButton/spawnVolumeModal/SizeSelector.tsx [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+export const SizeSelector: React.FC<Props> = ({ value, onChange }) => (
+  <SectionContainer>
+    <SectionLabel weight="medium">Volume Size</SectionLabel>
+    <ModalContent>
+      <InputLabel htmlFor="volumeSize">Size</InputLabel>
+      <InputNumber
+        data-cy="volumeSize"
+        id="volumeSize"
+        min={1}
+        max={500}
```
Comment (khelif96) [inline src/pages/spawn/spawnVolume/spawnVolumeButton/spawnVolumeModal/SizeSelector.tsx:20]: Users are able to create volumes that are larger then 500gb

File: src/pages/spawn/spawnVolume/spawnVolumeButton/spawnVolumeModal/TypeSelector.tsx [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+          <Option value={t} key={`type_option_${t}`}>
+            {t}
+          </Option>
+        ))}
+      </Select>
+    </ModalContent>
+  </SectionContainer>
+);
+
+const types = ["gp2", "io1", "sc1", "st1", "standard"];
```
Comment (khelif96) [inline src/pages/spawn/spawnVolume/spawnVolumeButton/spawnVolumeModal/TypeSelector.tsx:36]: Lets save this to the constants folder.

**Review by khelif96 — CHANGES_REQUESTED** [review]

File: src/components/Spawn/MountVolumeSelect.tsx (hunk 1 above)
Comment (SupaJoon) [reply src/components/Spawn/MountVolumeSelect.tsx:59]: There was no specific reason for this. I will change key to id to make the code more readable.

File: src/components/styles/Layout.tsx (hunk 2 above)
Comment (SupaJoon) [reply src/components/styles/Layout.tsx:68]: I prefer to keep this as to avoid an additional assignment that will only be used once in this file.

**Review by khelif96 — APPROVED** [review]

**Pull request merged**